bftengine/src/bftengine/ReplicaImp.hpp

```diff
@@ -398,10 +398,12 @@ class ReplicaImp : public InternalReplicaApi, public ReplicaForStateTransfer {
   SeqNum getPrimaryLastUsedSeqNum() const override { return primaryLastUsedSeqNum; }
   uint64_t getRequestsInQueue() const override { return requestsQueueOfPrimary.size(); }
   SeqNum getLastExecutedSeqNum() const override { return lastExecutedSeqNum; }
-  std::pair<PrePrepareMsg*, bool> buildPrePrepareMessage() override;
+  std::pair<std::unique_ptr<PrePrepareMsg>, bool> buildPrePrepareMessage() override;
```

Comment:
if this pair repeats more than 2-3 times, and seems like it does, I suggest to add alias decleration for it (call it PrePrepareMsgCreationResult or  PrePrepareMsgCreationPair ) or something like that